src/NuGet.Core/NuGet.Packaging/PackagesConfigWriter.cs

```diff
@@ -27,6 +27,36 @@ public class PackagesConfigWriter : IDisposable
         /// <summary>
         /// Create a packages.config writer
         /// </summary>
+        /// <param name="fullPath">the full path to write the XML packages.config file into, or load existing packages.config from</param>
+        /// <param name="createNew">Whether to create a new packages.config file, or load an existing one</param>
+        public PackagesConfigWriter(string fullPath, bool createNew)
+            : this(DefaultFrameworkNameProvider.Instance, fullPath, createNew)
+        {
+        }
+
+        public PackagesConfigWriter(IFrameworkNameProvider frameworkMappings, string fullPath, bool createNew)
+        {
+            _closed = false;
+            _frameworkMappings = frameworkMappings;
+
+            if (createNew)
+            {
+                _xDocument = new XDocument();
+                EnsurePackagesNode();
```

Comment:
Yes, that's what I thought. For existing packages.config, it should not try to fix the corrupted packages.config by adding the <packages /> node. We should throw an exception. I'll add that.